src/lxc/attach.c

```diff
@@ -873,7 +873,7 @@ static int attach_child_main(struct attach_clone_payload *payload)
 	}
 
 	ret = lxc_setgroups(0, NULL);
-	if (ret < 0)
+	if (ret < 0 && errno != EPERM)
```

Comment:
Sorry to cause more work - but I do think it's worth adding a debug msg in the case where it failed with EPERM.

On the one hand we don't want to completely break containers who are under deny_setgroup.  OTOH if i'm attaching to a container and it unexpectedly fails to drop some groups that i didn't want in the container, that's actually a big deal i guess.

Reply:
Back when I implemented the `lxc_setgroups()` function I made it such that it always logs :):
```
/* Simple covenience function which enables uniform logging. */
int lxc_setgroups(int size, gid_t list[])
{
	if (setgroups(size, list) < 0) {
		SYSERROR("Failed to setgroups().");
		return -errno;
	}
	NOTICE("Dropped additional groups.");

	return 0;
}
```